c2cgeoportal/views/tinyowsproxy.py

```diff
@@ -0,0 +1,190 @@
+        return self.settings.get("tinyows_url")
+
+    @view_config(route_name="tinyowsproxy")
+    def proxy(self):
+        self.user = self.request.user
+
+        if self.user is None:
+            raise HTTPUnauthorized(
+                "Authentication required",
+                headers=[("WWW-Authenticate", 'Basic realm="TinyOWS"')])
+        self.role_id = None if self.user is None else self.user.role.id
+
+        # params hold the parameters we're going to send to TinyOWS
+        params = dict(self.request.params)
+        self.lower_params = self._get_lower_params(params)
+
+        operation = self.lower_params.get("request", None)
+        typenames = \
+            set([normalize_typename(self.lower_params.get("typename"))]) \
+            if "typename" in self.lower_params \
+            else set()
+
+        method = self.request.method
+        if method == "POST":
+            try:
+                (operation, typenames_post) = \
+                    self._parse_body(self.request.body)
+            except Exception as e:
+                log.error("Error while parsing POST request body")
+                log.exception(e)
+                raise HTTPBadRequest(
+                    "Error parsing the request (see logs for more details)")
+
+            typenames = typenames.union(typenames_post)
+
+        if operation is None or operation == "":
+            operation = "getcapabilities"
+
+        if operation == "describefeaturetype":
+            # for DescribeFeatureType we require that exactly one type-name
+            # is given, otherwise we would have to filter the result
+            if len(typenames) != 1:
+                raise HTTPBadRequest(
+                    "Exactly one type-name must be given for " +
+                    "DescribeFeatureType requests")
+
+        if not self._is_allowed(typenames):
+            raise HTTPForbidden(
+                "No access rights for at least one of the given type-names")
+
+        # we want clients to cache GetCapabilities and DescribeFeatureType req.
+        use_cache = method == "GET" and operation in \
+            (u"getcapabilities", u"describefeaturetype")
+
+        response = self._proxy_callback(
+            operation=operation,
+            url=self._get_wfs_url(), params=params, cache=use_cache,
+            headers=self._get_headers(), body=self.request.body,
+            role_id=self.role_id
+        )
+        if use_cache:
+            response.cache_control.public = False
+            response.cache_control.private = True
+        return response
+
+    def _is_allowed(self, typenames):
+        """
+        Checks if the current user has the rights to access the given
+        type-names.
+        """
+        writable_layers = get_writable_layers(self.role_id)
+        return typenames.issubset(writable_layers)
+
+    def _get_headers(self):
+        headers = Proxy._get_headers(self)
+        if self.settings.get("tinyows_host", "") != "":
```

Comment:
> I say again:
1. We don't need the second test

Thanks for repeating this, again. But you are again missing an argument why it's not needed. Like I said before, I am testing for a non-empty string, because it doesn't make sense to set an empty string as host name.

But hey, because it's only used in debug mode, I made the change. For you. :heart: 

> 2. It's not elegant

This is your opinion. I am happy to learn a more elegant way to test if there is a non-empty string for a key in the settings.